python/GafferUI/Menu.py

```diff
@@ -405,11 +408,31 @@ def __evaluateItemValue( self, itemValue ) :
 
 		return itemValue
 
+	@GafferUI.LazyMethod()
```

Comment:
Why do we need to repolish lazily? Typically LazyMethod is for stuff that is either crazy slow, or for when multiple calls can be compressed into one, or for calls need to be deferred until a Widget is visible or playback stops. I'm surprised that any of that applies here...

Reply:
Calling repolish for every typed character slows down the menu update quite substantially. Is there an other way to defer the update or speed up the repolish?

Reply:
I would suggest :

- Don't repolish unless the error property has actually been changed (see https://github.com/GafferHQ/gaffer/blob/master/python/GafferUI/TextWidget.py#L100-L104). This means we will rarely repolish at all.
- Only repolish the search widget, not the whole menu.
- Put the property management and the repolish alongside each other in the same utility function, so it is clear what the repolish is for.